src/Mvc/Mvc.Api.Analyzers/src/AddResponseTypeAttributeCodeFixAction.cs

```diff
@@ -121,40 +121,18 @@ protected override async Task<Document> GetChangedDocumentAsync(CancellationToke
                 }
             }
 
-            if (root == null)
-            {
-                throw new ArgumentNullException(nameof(root));
-            }
-
             return document.WithSyntaxRoot(root);
         }
 
         private async Task<CodeActionContext?> CreateCodeActionContext(CancellationToken cancellationToken)
         {
             var root = await _document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-
-            if (root == null)
-            {
-                throw new ArgumentNullException(nameof(root));
-            }
-
             var semanticModel = await _document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var methodReturnStatement = (ReturnStatementSyntax)root.FindNode(_diagnostic.Location.SourceSpan);
             var methodSyntax = methodReturnStatement.FirstAncestorOrSelf<MethodDeclarationSyntax>();
             var method = semanticModel.GetDeclaredSymbol(methodSyntax, cancellationToken);
 
-            if (semanticModel == null)
-            {
-                throw new ArgumentNullException(nameof(semanticModel));
-            }
-
             var statusCodesType = semanticModel.Compilation.GetTypeByMetadataName(ApiSymbolNames.HttpStatusCodes);
-
-            if (statusCodesType == null)
-            {
-                throw new ArgumentNullException(nameof(statusCodesType));
-            }
-
```

Comment:
I assume you're removing all these null checks because some upstream checks (plus nullability?) tells us they are redundant. If it's meant to be more of a behavioral change please clarify.

Reply:
The code fix is only triggered because we correctly detected a valid code block to add a diagnostic to. Comparing notes with roslyn-analyzers, they treat these values to be well-defined in this case, and don't perform any checks